### PR TITLE
create index by alias if exists

### PR DESCRIPTION
**Pull request by KiraPC**

### CODE REVIEW

**Comment by KiraPC** [comment]
Fix for https://github.com/roman-right/beanie/issues/96

File: beanie/odm/utils/collection.py [hunk 1]
```diff
@@ -65,7 +65,13 @@ async def collection_factory(
     # Indexed field wrapped with Indexed()
     found_indexes = [
         IndexModel(
-            [(fname, fvalue.type_._indexed[0])], **fvalue.type_._indexed[1]
+            [
+                (
+                    fvalue.alias if hasattr(fvalue, "alias") else fname,
```
Comment (roman-right) [inline beanie/odm/utils/collection.py:70]: I think, here you can always use `fvalue.alias` . Pydantic already picks the right one there. Look:
https://github.com/samuelcolvin/pydantic/blob/master/pydantic/fields.py#L350

File: tests/odm/documents/test_init.py [hunk 2]
```diff
@@ -59,6 +60,23 @@ async def test_flagged_index_creation():
     }
 
 
+async def test_flagged_index_creation_with_alias():
```
Comment (roman-right) [inline tests/odm/documents/test_init.py:63]: That's great that you made a test for this!

**Review by roman-right — COMMENTED** [review]
Probably you can avoid extra check when you are choosing a name or alias. Just pick alias each time

File: beanie/odm/utils/collection.py (hunk 1 above)
Comment (KiraPC) [reply beanie/odm/utils/collection.py:70]: Can we assume that fvalue is always a Field instance?

File: tests/odm/documents/test_init.py (hunk 2 above)
Comment (KiraPC) [reply tests/odm/documents/test_init.py:63]: Never commit without tests :D 

(except when I added the typing, never tested type before in python :))

File: beanie/odm/utils/collection.py (hunk 1 above)
Comment (roman-right) [reply beanie/odm/utils/collection.py:70]: I think yes, as we iterate over `__fields__` here which is a dictionary with `ModelField` values only:
https://github.com/samuelcolvin/pydantic/blob/master/pydantic/main.py#L276

Comment (KiraPC) [reply beanie/odm/utils/collection.py]: Done!

**Pull request merged**